js/lib/urlutil.js

```diff
@@ -470,7 +470,10 @@ const UrlUtil = {
       if (parsed.protocol === 'about:') {
         return [parsed.protocol, parsed.path].join('')
       }
-      return parsed.origin.replace(/\/+$/, '')
+      if (parsed.origin.length !== 0) {
+        return parsed.origin.replace(/\/+$/, '')
+      }
+      return parsed.href.replace(/\/+$/, '')
```

Comment:
is this change still needed? i don't think this works on arbitrary URLs, ex:

```
muon.url.parse('sms://bing/kfdk').href.replace(/\/+$/, '')  // returns "sms://bing/kfdk"
```

Reply:
let's remove this diff if it's no longer needed

Reply:
This is still needed, `muon.url.parse` for `muon.url.parse('sms://bing/kfdk')` returns NULL origin. The warning message if this change is not present, the warning message will read `Allow Brave Browser to open an external application?`

Reply:
@jumde what if instead of this change, you just change https://github.com/brave/browser-laptop/pull/13475/files#diff-7b5b041c814805dd8496488bc6a53d3cR469 to `parsed.origin != undefined` so that the null origin will trigger https://github.com/brave/browser-laptop/pull/13475/files#diff-7b5b041c814805dd8496488bc6a53d3cR479 ?

also please add a test case in the getOrigin unit tests

Reply:
Host is also null for arbitrary URLs:

```
Url: hps://nodejs.org/en/download/
{ hash: '',
  hostname: '',
  host: '',
  href: 'hps://nodejs.org/en/download/',
  path: '//nodejs.org/en/download/',
  pathname: '//nodejs.org/en/download/',
  port: '',
  protocol: 'hps:',
  query: '',
  search: '',
  origin: '' }```

Reply:
@jumde i see. what if instead of changing the behavior of `getOrigin`, we just change https://github.com/brave/browser-laptop/pull/13475/files#diff-8a9bac13ef6264feb2b6da7c18d86b3cR500 such to `requestingOrigin || requestingUrl || 'Brave Browser'` so it displays correctly?

i am reluctant to change getOrigin since it's used in various places for security checks